dask/layers.py

```diff
@@ -409,11 +388,18 @@ def __init__(
         # much earlier.
         #
         # See https://github.com/dask/dask/pull/6051 for a detailed discussion.
-        self.annotations = self.annotations or {}
-        if "priority" not in self.annotations:
-            self.annotations["priority"] = {}
-        self.annotations["priority"]["__expanded_annotations__"] = None
-        self.annotations["priority"].update({_key: 1 for _key in self.get_split_keys()})
+        annotations = annotations or {}
+        self._split_keys = None
+        if "priority" not in annotations:
+            annotations["priority"] = self._key_priority
+
+        self._split_keys_set = set(self.get_split_keys())
+        super().__init__(annotations=annotations)
+
+    def _key_priority(self, key):
+        if key in self._split_keys_set:
+            return 1
+        return 0
```

Comment:
It's worth pointing out that an earlier version of this caused a regression earlier see https://github.com/dask/dask/pull/9994
and https://github.com/dask/dask/pull/10041 with the fix and a reason why this wasn't merged, yet. It accidentally triggered materialization, no longer an issue with this PR

Reply:
Thanks for the explanation. I'm just learning now that annotations can be callable - cool :)